src/EnergyPlus/OutputProcessor.cc

```diff
@@ -5091,311 +3593,317 @@ void UpdateDataandReport(EnergyPlusData &state, OutputProcessor::TimeStepType co
 
     if (EndTimeStepFlag) {
-        if (state.dataResultsFramework->resultsFramework->timeSeriesEnabled()) {
-            if (!state.dataResultsFramework->resultsFramework->RITimestepTSData.rVariablesScanned()) {
-                state.dataResultsFramework->resultsFramework->initializeRTSDataFrame(
-                    ReportingFrequency::TimeStep, op->RVariableTypes, op->NumOfRVariable);
-            }
-            if (!state.dataResultsFramework->resultsFramework->RITimestepTSData.iVariablesScanned()) {
-                state.dataResultsFramework->resultsFramework->initializeITSDataFrame(
-                    ReportingFrequency::TimeStep, op->IVariableTypes, op->NumOfIVariable);
-            }
-            state.dataResultsFramework->resultsFramework->RITimestepTSData.newRow(state.dataEnvrn->Month,
-                                                                                  state.dataEnvrn->DayOfMonth,
-                                                                                  state.dataGlobal->HourOfDay,
-                                                                                  op->TimeValue.at(TimeStepType::Zone).CurMinute,
-                                                                                  state.dataGlobal->CalendarYear);
-        }
-
-        for (TimeStepType thisTimeStepType : {TimeStepType::Zone, TimeStepType::System}) { // Zone, HVAC
-            for (int Loop = 1; Loop <= op->NumOfRVariable; ++Loop) {
-                if (op->RVariableTypes(Loop).timeStepType != thisTimeStepType) continue;
-                auto &rVar = op->RVariableTypes(Loop).VarPtr;
-                // Update meters on the TimeStep  (Zone)
-                if (rVar.MeterArrayPtr != 0 && !state.dataOutputProcessor->MeterValue.empty()) {
-                    Real64 TimeStepValue = rVar.TSValue * rVar.ZoneMult * rVar.ZoneListMult;
-                    for (int i = 1; i <= op->VarMeterArrays(rVar.MeterArrayPtr).NumOnMeters; i++) {
-                        int index = op->VarMeterArrays(rVar.MeterArrayPtr).OnMeters(i);
-                        state.dataOutputProcessor->MeterValue(index) += TimeStepValue;
+        if (rf->timeSeriesEnabled()) {
+            if (!rf->freqTSData[(int)ReportFreq::TimeStep].variablesScanned()) {
+                rf->initializeTSDataFrame(ReportFreq::TimeStep, op->outVars);
+            }
+            rf->freqTSData[(int)ReportFreq::TimeStep].newRow(state.dataEnvrn->Month,
+                                                               state.dataEnvrn->DayOfMonth,
+                                                               state.dataGlobal->HourOfDay,
+                                                               op->TimeValue[(int)TimeStepType::Zone].CurMinute,
+                                                               state.dataGlobal->CalendarYear);
+        }
+
+
+        // Update meters on the TimeStep  (Zone)
+        if (op->meterValues.capacity() > 0) {
```

Comment:
This update is driven by the meters not the variables.  Meters and variables are cross-linked during construction.